src/com/spikes2212/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,64 @@
+package com.spikes2212.robot.subsystems;
+
+import com.spikes2212.dashboard.ConstantHandler;
+import com.spikes2212.genericsubsystems.BasicSubsystem;
+import com.spikes2212.genericsubsystems.limitationFunctions.TwoLimits;
+import edu.wpi.first.wpilibj.Encoder;
+
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public class Elevator extends BasicSubsystem {
+
+    public static final Supplier<Double> SPEED = ConstantHandler.addConstantDouble("Elevator-Speed", 0.2);
+    public static final Supplier<Double> MIDDLE_SET_POINT = ConstantHandler.addConstantDouble(
+            "Lift-Middle-Position", 0
+    );
+
+
+
+    private Encoder encoder;
+
+    private Supplier<Boolean> maxLimit, minLimit;
+
+    public enum ElevatorState {
+         DOWN(0), MIDDLE_DOWN(1), MIDDLE(2), MIDDLE_UP(3), UP(4);
+
+        private int index;
+        ElevatorState(int index){
+            this.index = index;
+        }
+        public int getIndex(){
+            return index;
+        }
+    }
+
+
+    public Elevator(Consumer<Double> speedConsumer, Supplier<Boolean> maxLimit, Supplier<Boolean> minLimit,
+                    Encoder encoder) {
+        super(speedConsumer, new TwoLimits(maxLimit, minLimit));
+        this.encoder = encoder;
+        this.maxLimit = maxLimit;
+        this.minLimit = () -> {
+            if (minLimit.get()) resetEncoder();
+            return minLimit.get();
+        };
+
+    }
+
+    public ElevatorState getPosition() {
+        if (encoder.get() == MIDDLE_SET_POINT.get())
```

Comment:
call it getState otherwise it will be very confusing.
sorry for the trouble btw (;